surrealdb/http.py

```diff
@@ -203,13 +203,13 @@ async def select(self, thing: str) -> List[Dict[str, Any]]:
             Select a specific record from a table (or other entity)
                 person = await db.select('person:h5wxrf2ewk8xjxosxtyc')
         """
-        if ":" in thing:
-            table, id = thing.split(":")
-            response = await self._request(method="GET", uri=f"/key/{table}/{id}")
-            if not response:
-                raise SurrealException(f"Key {id} not found in table {table}")
-        else:
-            response = await self._request(method="GET", uri=f"/key/{thing}")
+        table, record_id = thing.split(":") if ":" in thing else (thing, None)
```

Comment:
I don't like this line very much. It's a confusing approach to a simple problem. In my opinion the whole function should be refactored so that there is no need for a split combined with a tuple unpacking combined with a ternary operator. It is confusing and is trying to put a dynamic aspect into something which shouldn't be dynamic.

Reply:
I understand that opinions may vary, but I believe that the code you referred to as disliked is actually a Pythonic way of splitting a string into two parts using a colon ":" as a separator. The code employs tuple unpacking to assign the result of the split function to two variables, table and record_id.

Overall, this approach is concise and readable, making it a common and preferred way to handle the scenario where an input string may or may not contain a separator character.

Reply:
I understand where you are coming from @CaedenPH, that is a valid opinion.
Generally, functions should do one task. The question is, how do you define one task?
The way we have defined it here is that the one task that it does is sending one CRUD instruction to the database based on an input string.

This aligns with how things are in the WebSocket implementation and the implementation in other libraries such as the JavaScript one.

I agree with @HiteshGorana that the suggested approach can make the code more concise and easier to read.

The function signature wouldn't really change if you split this into two functions, it would still accept one string and return a dict. The doc string has examples for what is an acceptable input, which would be split then between two functions. This seems to me like it would be unnecessary duplication 🤔 

There might be something I'm missing though. Happy for you to create another PR with your recommendation and we can discuss there.